lab_1/dictionary_test.py

```diff
@@ -0,0 +1,214 @@
+        """
+        bad_inputs = ['string', (), None, 9, 9.34, True]
+        for bad_input in bad_inputs:
+            self.assertEqual(get_top_n_words(bad_input, 2), [])
+
+    def test_get_top_n_words_empty(self):
+        """
+        Get top number of words with empty arguments
+        """
+        self.assertEqual(get_top_n_words({}, 10), [])
+
+    def test_get_top_n_words_incorrect_numbers(self):
+        """
+        Get top number of words using incorrect number of words parameter
+        """
+        self.assertEqual(get_top_n_words({}, -1), [])
+        self.assertEqual(get_top_n_words({'happy': 2}, 0), [])
+
+
+class GetConcordanceTest(unittest.TestCase):
+    """
+    Tests get concordance function
+    """
+
+    def test_get_concordance_ideal(self):
+        """
+        Ideal get concordance scenario
+        """
+        self.assertEqual(get_concordance(['the', 'weather', 'is', 'sunny', 'the', 'man', 'is', 'happy',
+                                          'the', 'dog', 'is', 'glad', 'but', 'the', 'cat', 'is', 'sad'],
+                                         'happy', 2, 3), ['man is happy the dog is'])
+
+    def test_get_concordance_several_contexts(self):
+        """
+        Checks that a concordance list can be created for several contexts
+        """
+        self.assertEqual(get_concordance(['the', 'weather', 'is', 'sunny', 'the', 'man', 'is', 'happy',
+                                          'the', 'dog', 'is', 'happy', 'but', 'the', 'cat', 'is', 'sad'],
+                                         'happy', 2, 3), ['man is happy the dog is', 'dog is happy but the cat'])
+
+    def test_get_concordance_empty_inputs(self):
+        """
+        Checks that function can handle empty argument inputs
+        """
+        self.assertEqual(get_concordance([], 'happy', 2, 3), [])
+        self.assertEqual(get_concordance(['happy'], '', 2, 3), [])
+        self.assertEqual(get_concordance(['happy'], 'happy', 0, 0), ['happy'])  # ?
+
+    def test_get_concordance_bad_number_inputs(self):
+        """
+        Checks that function can handle incorrect number inputs
+        """
+        self.assertEqual(get_concordance(['happy', 'man'], 'happy', -1, 0), ['happy'])
+        self.assertEqual(get_concordance(['happy', 'man'], 'happy', 0, 1), ['happy man'])
+        self.assertEqual(get_concordance(['happy', 'man'], 'man', -1, 0), ['man'])
+        bad_inputs = [[], {}, 'string', (), None, 9.34, True]
+        for bad_input in bad_inputs:
+            self.assertEqual(get_concordance(['happy', 'man', 'went'], 'man', bad_input, bad_input), [])
+
+    def test_get_concordance_bad_inputs(self):
+        """
+        Checks that function can handle incorrect inputs
+        """
+        bad_inputs = [{}, 'string', (), None, 9, 9.34, True]
+        for bad_input in bad_inputs:
+            self.assertEqual(get_concordance(bad_input, 'happy', 2, 3), [])
+        bad_inputs = [{}, [], (), None, 9, 9.34, True]
```

Comment:
May be concatenate them in one so there is no code duplication and unnecessary assigning?

```suggestion
        bad_inputs = [{}, [], 'string', (), None, 9, 9.34, True]
        for bad_input in bad_inputs:
            self.assertEqual(get_concordance(bad_input, 'happy', 2, 3), [])
```

Reply:
yep, it is possible to avoid duplication, but `bad_inputs` variable changes a little due to argument testing, e.g `'string'` is not allowed for the first argument, but allowed for the second. I'll refactor this test so, please, check it again then.